core/src/main/java/com/linecorp/armeria/common/metric/CounterGauge.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.concurrent.atomic.LongAdder;
+
+public class CounterGauge extends AbstractGauge {
+
+    private final LongAdder counter = new LongAdder();
+
+    public CounterGauge(MetricKey key, MetricUnit unit, String description) {
+        super(key, unit, description);
+    }
+
+    @Override
+    public long value() {
+        return counter.sum();
```

Comment:
Talked offline and he was somewhat concerned about `value()` returning a negative value. Let me the returned value is wrapped around so it does not return a negative value.